src/EditorFeatures/CSharp/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -16,5 +16,38 @@ protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, C
         {
             return symbol;
         }
+
+        protected override bool TokenIsPartOfDeclaringSyntax(SyntaxToken token)
+        {
+            var parentNode = token.Parent;
+            switch (parentNode.Kind())
+            {
+                case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.ConstructorDeclaration:
+                case SyntaxKind.MethodDeclaration:
+                case SyntaxKind.PropertyDeclaration:
+                case SyntaxKind.EventDeclaration:
+                case SyntaxKind.FieldDeclaration:
+                case SyntaxKind.GetAccessorDeclaration:
+                case SyntaxKind.SetAccessorDeclaration:
+                case SyntaxKind.AddAccessorDeclaration:
+                case SyntaxKind.RemoveAccessorDeclaration:
+                case SyntaxKind.VariableDeclaration:
+                case SyntaxKind.DelegateDeclaration:
+                case SyntaxKind.EnumDeclaration:
+                case SyntaxKind.StructDeclaration:
+                case SyntaxKind.Parameter:
+                case SyntaxKind.EnumMemberDeclaration:
+                case SyntaxKind.VariableDeclarator:
+                case SyntaxKind.LocalFunctionStatement:
+                case SyntaxKind.TypeParameter:
+                case SyntaxKind.AnonymousObjectCreationExpression:
+                    return true;
+                case SyntaxKind.IdentifierName:
+                    return parentNode?.Parent.IsKind(SyntaxKind.NameColon, SyntaxKind.NameEquals) ?? false;
```

Comment:
:memo: This expression is one of the ones that led me to search for an alternative solution like I asked about above (even if it only approximates the original goal). In particular, there are cases of this form which could be considered a reference (e.g. to a parameter) and would expect Ctrl+Click to work. Rather than trying to verify that every case is correct, we could go with the approximate solution and not have to worry about it.